### PR TITLE
Add `claimsToHeaders` to internal ee ext-auth api

### PR DESCRIPTION
**Pull request by inFocus7**
# Description

Adding `claimsToHeaders` support to Gloo Edge (Enterprise).

## API changes
- Added `AccessToken` and `IdentityToken` (which include `claimsToHeaders`) to our internal API so we can translate it to ext-auth's 1:1 field.

# Context

This API was already introduced for OIDC [here](https://github.com/solo-io/gloo/pull/8594/files) + implemented in ext-auth [here](https://github.com/solo-io/ext-auth-service/pull/616/files). We are essentially copy-pasting/exposing it to our internal API for translation purposes in Gloo EE.

## Testing steps

I manually verified behavior in https://github.com/solo-io/solo-projects/pull/5440.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 1]
```diff
@@ -1566,6 +1566,36 @@ message ExtAuthConfig {
...
+            string claim = 1;
+
+            // The header to copy the claim to, such as `x-sub`.
+            string header = 2;
+
+            // If the header exists, append the claim value to the header (true), or overwrite any existing value (false). The default behavior is to overwrite any existing value (false).
+            bool append = 3;
+        }
+
+        // Optional: Configuration specific to the OAuth2 access token received and processed by the ext-auth-service.
```
Comment (sheidkamp) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1582]: Like how these are labelled "optional"

**Review by sheidkamp — APPROVED** [review]
LGTM. Was considering waiting to see e2e before approving, but since this is working in Platform and this is basically copying that API, I'm confident it will work.

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto (hunk 1 above)
Comment (inFocus7) [reply projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1582]: Thanks to @jjamroga, since I just copy-pasted lol

**Comment by inFocus7** [comment]
> considering waiting to see e2e before approving
I linked the EE repo which includes manual testing instructions + additions to our tests which have passed

**Review by sam-heilbron — APPROVED** [review]

**Review by sheidkamp — APPROVED** [review]

**Pull request merged**